src/components/UASTViewer.js

```diff
@@ -1,13 +1,19 @@
-import React from 'react';
+import React, { Component } from 'react';
 import styled from 'styled-components';
 
 const Container = styled.div`
 `
 
-export default function UASTViewer() {
-  return (
-    <Container>
-      UASTViewer
-    </Container>
-  );
+export default class UASTViewer extends Component {
+  selectNodes(start, end) {
```

Comment:
Pass only the cursor position (or "end of selection" in case of doing a selection)
Pay attention to the selection direction in order to provide the right cursor position.